test-plan/eventing/subscription-lifecycle-conformance.md

```diff
@@ -0,0 +1,257 @@
+
+```
+kubectl apply -f control-plane/subscription-lifecycle/services.yaml
+```
+
+## [Test] Subscription status update
+
+The `status.physicalSubscription` should have been updated with the 
+
+```
+kubectl get subscription conformance-subscription -ojsonpath="{.status.physicalSubscription}"
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscription-status-update"
+  "output": {
+	  "obtainedPhysicalSubscription": {
+      "deadLetterSinkUri": "<DEADLETTERSINK_URI>",
+      "subscriberUri": "<SUBSCRIBER_URI>"
+    }
+  }
+}
+```
+
+## [Test] Subscription Readiness 2
+
+Check for condition type `Ready` with status `False` cause with even with the Channel some of the preconditions on the spec are not met: 
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscription-readiness-2"
+  "output": {
+	  "expectedType": "Ready",
+	  "expectedStatus": "True"
+  }
+}
+```
+
+## [Test] Subscribtion with no Subscriber
+
+Lets try to create a Subscription that have no `spec.subscriber` defined: 
+
+```
+kubectl apply -f control-plane/subscription-lifecycle/subscription-no-subscriber.yaml
+```
+
+You should get the following error: 
+```
+Error from server (BadRequest): error when creating "control-plane/subscription-lifecycle/subscription-no-subscriber.yaml": admission webhook "validation.webhook.eventing.knative.dev" denied the request: validation failed: missing field(s): spec.reply, spec.subscriber
+the Subscription must reference at least one of (reply or a subscriber)
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscriptin-no-subscriber"
+  "output": {
+	  "expectedError": "<EXPECTED_ERROR>"
+  }
+}
+```
+
+## [Test] Subscribtion channel in different namespace
+
+Lets try to create a Subscription that have `spec.channel.namespace` defined and different from `""`: 
+
+```
+kubectl apply -f control-plane/subscription-lifecycle/subscription-channel-namespace.yaml
+```
+
+You should get the following error: 
+```
+Error from server (BadRequest): error when creating "control-plane/subscription-lifecycle/subscription-channel-namespace.yaml": admission webhook "validation.webhook.eventing.knative.dev" denied the request: validation failed: must not set the field(s): spec.channel.namespace
+only name, apiVersion and kind are supported fields
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscriptin-channel-different-namespace"
+  "output": {
+	  "expectedError": "<EXPECTED_ERROR>"
+  }
+}
+```
+
+# Clean up & Congratulations
+
+Make sure that you clean up all resources created in these tests by running (ignore the errors :p): 
+
+```
+kubectl delete -f control-plane/subscription-lifecycle/
```

Comment:
I think there are only 4 objects to delete so we could put up a one liner command to do so. What about adding a label to all objects at manifests, for example `app=conformance` and then cleaning by issuing a
```
kubectl delete all -l app=conformance
```
WDYT?